### PR TITLE
Optional output normalization

### PR DESCRIPTION
**Pull request by pzhanggit**
1. added a flag to output predictions with unit
2. saved minimum and maximum values used in normalization to data files *.pkl

### CODE REVIEW

File: examples/configuration.json [hunk 1]
```diff
@@ -13,5 +13,6 @@
     "output_dim": 1,
     "predicted_value_option": 1,
     "dataset_option": "CuAu_32atoms",
-    "model_type": "PNN"
+    "model_type": "PNN",
+    "out_wunit": "True"
```
Comment (streeve) [inline examples/configuration.json:17]: What about `"normalize_output"` (which would be the opposite)?

File: data_utils/serialized_dataset_loader.py [hunk 2]
```diff
@@ -48,6 +48,8 @@ def load_serialized_data(
         """
         dataset = []
         with open(dataset_path, "rb") as f:
+            x_minmax = pickle.load(f)
```
Comment (streeve) [inline data_utils/serialized_dataset_loader.py:51]: Do you use these values anywhere?

File: utils/utils.py [hunk 3]
```diff
@@ -207,6 +221,10 @@ def dataset_loading_and_splitting(
             distributed_data_parallelism=distributed_data_parallelism,
         )
     else:
+        raise ValueError(
+            "Not ready yet. Should re-normalize mixed datasets based on joint min_max",
```
Comment (streeve) [inline utils/utils.py:225]: "Chosen dataset option not yet supported", then maybe a `FIXME` comment about the needed re-normalizations

**Review by streeve — COMMENTED** [review]
Is this what Max needs to build the unit tests more easily?

File: data_utils/serialized_dataset_loader.py (hunk 2 above)
Comment (pzhanggit) [reply data_utils/serialized_dataset_loader.py:51]: Not used here, but not sure if we could skip the first two and load the third item directly

Comment (streeve) [reply data_utils/serialized_dataset_loader.py:51]: Makes sense

File: examples/configuration.json (hunk 1 above)
Comment (pzhanggit) [reply examples/configuration.json]: Fixed in @5948ba39433071174fd06a3ef9b9f90805546870

File: run_config_input.py [hunk 4]
```diff
@@ -185,6 +186,34 @@ def run_normal_config_file(config_file="./examples/configuration.json"):
...
+            config["y_minmax"].append(x_minmax[:, :, 1].tolist())
+        elif config["predicted_value_option"] == 5:
+            config["y_minmax"].append(y_minmax[:, 0].tolist())
+            config["y_minmax"].append(x_minmax[:, :, 2].tolist())
+        elif config["predicted_value_option"] == 6:
+            config["y_minmax"].append(y_minmax[:, 0].tolist())
+            config["y_minmax"].append(x_minmax[:, :, 1].tolist())
+            config["y_minmax"].append(x_minmax[:, :, 2].tolist())
+    else:
+        config["denormalize_output"] = "False"
```
Comment (streeve) [inline run_config_input.py:215]: I like the idea of setting defaults like this to not break compatibility with existing configurations

File: utils/utils.py (hunk 3 above)
Comment (pzhanggit) [reply utils/utils.py]: Fixed in 5948ba3

File: run_config_input.py [hunk 5]
```diff
@@ -14,6 +14,7 @@
     AtomFeatures,
     Dataset,
 )
+import pickle
 
 
 def run_normal_terminal_input():
```
Comment (streeve) [inline run_config_input.py:20]: Do we need changes here as well?

**Review by allaffa — APPROVED** [review]

**Pull request merged**